plexus-java/src/main/java/org/codehaus/plexus/languages/java/version/JavaClassVersion.java

```diff
@@ -0,0 +1,141 @@
+package org.codehaus.plexus.languages.java.version;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Reads the bytecode of a Java class to detect the major, minor and Java
+ * version that was compiled.
+ *
+ * @author Jorge Solórzano
+ */
+public final class JavaClassVersion {
```

Comment:
I'd prefer to use JavaClassfileVersion, to better match https://github.com/asotona/jdk/blob/2454c8e233bd96c77876ae0c6e2ecdc580e2271f/src/java.base/share/classes/jdk/internal/classfile/ClassfileVersion.java . So the methods majorVersion and minorVersion are already fine.